app/src/src_nonfreenet/org/breezyweather/sources/ekuk/EkukService.kt

```diff
@@ -0,0 +1,180 @@
+    }
+
+    override fun requestWeather(
+        context: Context,
+        location: Location,
+        requestedFeatures: List<SourceFeature>,
+    ): Observable<WeatherWrapper> {
+        val airQualityStation = location.parameters.getOrElse(id) { null }?.getOrElse("airQualityStation") { null }
+        val pollenStation = location.parameters.getOrElse(id) { null }?.getOrElse("pollenStation") { null }
+
+        val formatter = SimpleDateFormat("dd.MM.yyyy", Locale.ENGLISH)
+        formatter.timeZone = TimeZone.getTimeZone("Europe/Tallinn")
+        val now = Calendar.getInstance()
+        val today = formatter.format(now.time)
+        now.add(Calendar.DATE, -1)
+        val yesterday = formatter.format(now.time)
+
+        val failedFeatures = mutableListOf<SourceFeature>()
+        val airQuality = if (SourceFeature.AIR_QUALITY in requestedFeatures) {
+            if (!airQualityStation.isNullOrEmpty()) {
+                mApi.getObservations(
+                    station = airQualityStation,
+                    indicators = EKUK_AIR_QUALITY_INDICATORS,
+                    range = "$yesterday,$today"
+                ).onErrorResumeNext {
+                    failedFeatures.add(SourceFeature.AIR_QUALITY)
+                    Observable.just(emptyList())
+                }
+            } else {
+                // Do not fail: airQualityStation is empty if the nearest one is > 50km away
+                Observable.just(emptyList())
+            }
+        } else {
+            Observable.just(emptyList())
+        }
+
+        val pollen = if (SourceFeature.POLLEN in requestedFeatures) {
+            if (!pollenStation.isNullOrEmpty()) {
+                mApi.getObservations(
+                    station = pollenStation,
+                    indicators = EKUK_POLLEN_INDICATORS,
+                    range = "$yesterday,$today"
+                ).onErrorResumeNext {
+                    failedFeatures.add(SourceFeature.POLLEN)
+                    Observable.just(emptyList())
+                }
+            } else {
+                // Do not fail: pollenStation is empty if the nearest one is > 50km away
+                Observable.just(emptyList())
+            }
+        } else {
+            Observable.just(emptyList())
+        }
+
+        return Observable.zip(airQuality, pollen) {
+                airQualityResult: List<EkukObservationsResult>,
+                pollenResult: List<EkukObservationsResult>,
+            ->
+            convert(
+                airQualityResult = if (SourceFeature.AIR_QUALITY in requestedFeatures) {
+                    airQualityResult
+                } else {
+                    null
+                },
+                pollenResult = if (SourceFeature.POLLEN in requestedFeatures) {
+                    pollenResult
+                } else {
+                    null
+                },
+                failedFeatures = failedFeatures
+            )
```

Comment:
Delete the `convert()` function and replace directly with:
```
WeatherWrapper(
        airQuality = if (SourceFeature.AIR_QUALITY in requestedFeatures) { airQualityResult?.let { AirQualityWrapper(current = getAirQuality(it)) } },
        pollen = if (SourceFeature.POLLEN in requestedFeatures) { pollenResult?.let { PollenWrapper(current = getPollen(it)) } },
        failedFeatures = failedFeatures
    )
```
(Apply the correct lint after copying)

Reply:
Fixed